synapse/storage/databases/main/event_push_actions.py

```diff
@@ -12,6 +12,67 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+
+"""Responsible for storing and fetching push actions / notifications.
+
+There are two main uses for push actions:
+  1. Sending out push to a user's device; and
+  2. Tracking per-room per-user notification counts (used in sync requests).
+
+For the former we simply use the `event_push_actions` table, which contains all
+the calculated actions for a given user (which were calculated by the
+`BulkPushRuleEvaluator`).
+
+For the latter we could simply count the number of rows in `event_push_actions`
+table for a given room/user, but in practice this could be *very* heavyweight
+when there were a large number of notifications (due to e.g. the user never
+reading a room). Plus, keeping all push actions indefinitely uses a lot of disk
+space.
```

Comment:
We know it is heavyweight:

```suggestion
For the latter we could simply count the number of rows in `event_push_actions`
table for a given room/user, but in practice this is *very* heavyweight when there
were a large number of notifications (due to e.g. the user never reading a room).
Plus, keeping all push actions indefinitely uses a lot of disk space.
```